libtenzir/src/transfer.cpp

```diff
@@ -184,7 +184,7 @@ auto transfer::prepare(chunk_ptr chunk) -> caf::error {
 
 auto transfer::perform() -> caf::error {
   auto code = easy_.perform();
-  if (code == curl::easy::code::ok) {
+  if (code != curl::easy::code::ok) {
```

Comment:
No, it's too low-level to write something that may have occurred but not always.

Reply:
I don't think so: `save https <url that does not resolve>` did nothing before and signaled success instead of erroring. To me, that is a very user-facing bug.

Reply:
I have no strong opinion here, so do as you like.

Reply:
This can be easily triggered on the user side, so I added a changelog item in https://github.com/tenzir/tenzir/pull/4281/commits/0978fc38aa2668e38bb41980a7ac94c8ba7b491d.